### PR TITLE
[Merged by Bors] - sync: fail on missing input vector

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation

following up on discussion with @dshulyak on missing input vector during sync

## Changes

- sync fails on missing input vector
- layer fetcher tag layer as empty in mesh instead of syncer. this removes the need to use ErrZeroBlockLayer to signal empty layer from layer fetcher to syncer. syncer simply try to get layer from mesh after fetching is done.
- update syncer tests to be consistent with new logic where we don't fetch blocks for genesis layer anymore

## Test Plan

unit tests

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by countvonzero** [comment]
bors try

File: layerfetcher/layers.go [hunk 1]
```diff
@@ -354,39 +352,48 @@ func (l *Logic) receiveLayerContent(ctx context.Context, layerID types.LayerID,
...
 		if err == nil {
 			err = res.err
 		}
 	}
 	result := LayerPromiseResult{Layer: layerID}
+	// we tolerate errors from peers as long as we fetched all known blocks in this layer.
 	if missing || !success {
 		result.Err = err
 	}
+	if result.Err == nil && !hasBlocks { // none of the successful peers has data
```
Comment (dshulyak) [inline layerfetcher/layers.go]: we can use len(lyrResult.blocks) instead of one more variable

**Review by dshulyak — APPROVED** [review]

File: layerfetcher/layers.go (hunk 1 above)
Comment (countvonzero) [reply layerfetcher/layers.go]: done. thanks. i also moved saving input vector to the same block and report error on failure to save input vector in db

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**